### PR TITLE
Enhance resolution of histogram

### PR DESCRIPTION
**Pull request by t-bltg**
Fix https://github.com/JuliaPlots/UnicodePlots.jl/issues/124.

- [x] generate new ref imgs

Representative examples (thanks to @aminnj):
```julia
julia> using UnicodePlots, StatsBase
julia> UnicodePlots.histogram(Histogram(1:15., vcat(collect(1:13), 400)))
                ┌                                        ┐ 
   [ 1.0,  2.0) ┤▎ 1                                       
   [ 2.0,  3.0) ┤▎ 2                                       
   [ 3.0,  4.0) ┤▍ 3                                       
   [ 4.0,  5.0) ┤▍ 4                                       
   [ 5.0,  6.0) ┤▌ 5                                       
   [ 6.0,  7.0) ┤▌ 6                                       
   [ 7.0,  8.0) ┤▋ 7                                       
   [ 8.0,  9.0) ┤▋ 8                                       
   [ 9.0, 10.0) ┤▊ 9                                       
   [10.0, 11.0) ┤▊ 10                                      
   [11.0, 12.0) ┤▉ 11                                      
   [12.0, 13.0) ┤█▏ 12                                     
   [13.0, 14.0) ┤█▎ 13                                     
   [14.0, 15.0) ┤███████████████████████████████████ 400  
                └                                        ┘ 
                                Frequency
```

### CODE REVIEW

File: src/graphics/bargraphics.jl [hunk 1]
```diff
@@ -4,21 +4,23 @@ mutable struct BarplotGraphics{R<:Number} <: GraphicsArea
...
 
     function BarplotGraphics(
             bars::AbstractVector{R},
             char_width::Int,
             color::UserColorType,
-            symb::Union{Char,String},
-            transform) where {R}
-        length(symb) == 1 || throw(ArgumentError("The symbol to print has to be a single character, got: \"" * symb * "\""))
+            symbols::Union{S},
+            transform) where {R, S <: Union{Char, String}}
```
Comment (aminnj) [inline src/graphics/bargraphics.jl:15]: I was getting a crash trying to run
```julia
julia> using UnicodePlots

julia> UnicodePlots.histogram(randn(1000) * 0.1, closed = :right, nbins = 15)
```

until I did:
```diff
-            transform) where {R, S <: Union{Char, String}}
+            transform) where {R, S <: Union{Char, String, Vector{String}}}
```

But then I got
```
                  ┌                                        ┐
   (-0.35, -0.3 ] ┤▉ 2
   (-0.3 , -0.25] ┤█▏ 6
   (-0.25, -0.2 ] ┤██▎ 13
   (-0.2 , -0.15] ┤███████▎ 43
   (-0.15, -0.1 ] ┤██████████████████▏ 104
   (-0.1 , -0.05] ┤███████████████████████████▏ 153
   (-0.05,  0.0 ] ┤███████████████████████████████▏ 174
   ( 0.0 ,  0.05] ┤███████████████████████████████████▏ 194
   ( 0.05,  0.1 ] ┤████████████████████████████▏ 158
   ( 0.1 ,  0.15] ┤█████████████████▏ 97
   ( 0.15,  0.2 ] ┤█████▏ 30
   ( 0.2 ,  0.25] ┤███▎ 20
   ( 0.25,  0.3 ] ┤▉ 5
   ( 0.3 ,  0.35] ┤▉ 1
                  └                                        ┘
```
should the bars for 1, 2, 5, 6 be that close?

Comment (t-bltg) [reply src/graphics/bargraphics.jl:15]: Thanks for the feedback, yes that PR is not yet ready, needs more polishing, tests, ...
About the symbols being close, no I think my maths on this one are wrong :wink:

File: src/graphics/bargraphics.jl [hunk 2]
```diff
@@ -50,16 +52,23 @@ end
...
     max_bar_width = max(g.char_width - 2 - g.max_len, 1)
     val = g.transform(bar)
-    bar_len = max_freq > 0 ? round(Int, max(val, zero(val))/max_freq * max_bar_width, RoundNearestTiesUp) : 0
-    bar_str = max_freq > 0 ? repeat(g.symb, bar_len) : ""
+    frac = max_freq > 0 ? max(val, zero(val)) / max_freq : 0
+    bar_head = round(Int, frac * max_bar_width, RoundDown)
+    print_color(g.color, io, max_freq > 0 ? repeat(g.symbols[nsyms], bar_head) : "")
+    if nsyms > 1 && frac > 0
+        syms_tail = g.symbols[1:(nsyms-1)]
+        rem = (frac - bar_head / max_bar_width) / frac * (length(syms_tail) - 1)
```
Comment (aminnj) [inline src/graphics/bargraphics.jl:65]: `rem = (frac*max_bar_width - bar_head) * (length(syms_tail) - 1)` seems to give more realistic bars

```julia
julia> using UnicodePlots, StatsBase

julia> UnicodePlots.histogram(Histogram(1.0:8,[0,2,4,6,8,10,400])) # before
              ┌                                        ┐
   [1.0, 2.0) ┤ 0
   [2.0, 3.0) ┤▉ 2
   [3.0, 4.0) ┤▉ 4
   [4.0, 5.0) ┤▉ 6
   [5.0, 6.0) ┤▉ 8
   [6.0, 7.0) ┤▉ 10
   [7.0, 8.0) ┤███████████████████████████████████▏ 400
              └                                        ┘
                              Frequency

julia> UnicodePlots.histogram(Histogram(1.0:8,[0,2,4,6,8,10,400])) # after
              ┌                                        ┐
   [1.0, 2.0) ┤ 0
   [2.0, 3.0) ┤▎ 2
   [3.0, 4.0) ┤▍ 4
   [4.0, 5.0) ┤▌ 6
   [5.0, 6.0) ┤▋ 8
   [6.0, 7.0) ┤▊ 10
   [7.0, 8.0) ┤███████████████████████████████████▏ 400
              └                                        ┘
```

File: src/graphics/bargraphics.jl (hunk 1 above)
Comment (aminnj) [reply src/graphics/bargraphics.jl:15]: Ah, ok, see my comment below. Hope it helps :)

Comment (t-bltg) [reply src/graphics/bargraphics.jl:15]: @aminnj, about the types (`Union{Char, String, Vector{String}}`) on what julia version are you ? 
It works with the original PR on my machine using `julia@1.6.2`, can't reproduce the crash ...

Comment (aminnj) [reply src/graphics/bargraphics.jl:15]: I was using 1.6.0. I'll try again with 1.6.2 tomorrow and see if that fixes it.

Comment (t-bltg) [reply src/graphics/bargraphics.jl:15]: We'll see anyway if this PR passes on older releases in CI when I will have generated and pushed new ref imgs.

File: src/graphics/bargraphics.jl [hunk 3]
```diff
@@ -30,8 +32,8 @@ function BarplotGraphics(
         char_width::Int,
         transform = identity;
         color::UserColorType = :green,
-        symb = "■") where {R <: Number}
-    BarplotGraphics(bars, char_width, crayon_256_color(color), symb, transform)
+        symbols = ["■"]) where {R <: Number}
+    BarplotGraphics(bars, char_width, crayon_256_color(color), symbols, transform)
```
Comment (johnnychen94) [inline src/graphics/bargraphics.jl:36]: Changing keyword names can be breaking so I think we should deprecate it for every place...

```suggestion
        symb = nothing, # deprecated
        symbols = ["■"]) where {R <: Number}
    if !isnothing(symb)
        depwarn("keyword \"symb\" is deprecated in favor of \"symbols\"", :BarplotGraphics)
        symbols = [symb]
    end
    BarplotGraphics(bars, char_width, crayon_256_color(color), symbols, transform)
```

File: README.md [hunk 4]
```diff
@@ -257,9 +257,9 @@ parameters.
 
     The type of canvas that should be used for drawing (see section "Low-level Interface")
 
-- `symb::AbstractString = "▪"`:
+- `symbols::AbstractString = ["▪"]`:
```
Comment (johnnychen94) [inline README.md:260]:
```suggestion
- `symbols::AbstractArray{String} = ["▪"]`:
```

**Review by johnnychen94 — APPROVED** [review]
LGTM, I believe you know more details on this.

**Comment by t-bltg** [comment]
> The only concern is its breakage.

Added `_handle_deprecated_symb`.
Added `@test_logs` (depwarn) tests.

**Pull request merged**